code/renderers/svelte/src/render.ts

```diff
@@ -24,6 +26,20 @@ function teardown(canvasElement: SvelteRenderer['canvasElement']) {
   componentsByDomElement.delete(canvasElement);
 }
 
+/**
+ * This is a workaround for the issue that when resetting args,
+ * the story needs to be remounted completely to revert to the component's default props.
+ * This is because Svelte does not itself revert to defaults when a prop is undefined.
+ * See https://github.com/storybookjs/storybook/issues/21470#issuecomment-1467056479
+ *
+ * We listen for the RESET_STORY_ARGS event and store the storyId to be reset
+ * We then use this in the renderToCanvas function to force remount the story
+ */
+const storyIdsToRemountFromResetArgsEvent = new Set<string>();
```

Comment:
I'm unsure if this Set would ever have more than one entry at a time. It would only occur if `RESET_STORY_ARGS` were sent for multiple stories at the same time, and their `renderToCanvas` were all called after them (because `renderToCanvas` immediately removes the story id from the Set again).

I can't think of a scenario where that would happen - in that sense this could be simplified to be a single string instead of a Set.

However this feels more bulletproof in a way.

Reply:
Yeah I agree with all that 💯